php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2835,8 +2889,32 @@ abstract class NDB_BVL_Instrument extends NDB_Page
         if ($request->getMethod() === "POST" && !isset($req['ClearInstrument'])) {
             $this->save();
         }
-        return (new \Laminas\Diactoros\Response())
+
+        // Disable form if data entry is complete
+        $user      = $request->getAttribute("user");
+        $sessionID = $request->getQueryParams()["sessionID"];
+        $timepoint = \Timepoint::singleton($sessionID);
+
+        // create an instrument status object
+        $status = new \NDB_BVL_InstrumentStatus;
+        $status->select($this->commentID);
+
+        // freeze the form to prevent data entry
+        if ($status->getDataEntryStatus() != 'In Progress'
+            || !$user->hasPermission('data_entry')
+            || (isset($timepoint)
+            && !in_array(
+                $timepoint->getData("CenterID"),
+                $user->getData("CenterIDs")
```

Comment:
please add the third argument `true` to the in_array function to avoid unexpected behaviour

Reply:
I think `$user->hasCenter($timepoint->getCenterID())` would be better so that it doesn't need to change after #7359 (and it's more explicit what it's trying to do..)